### DISCUSSION TITLE
TX81Z not receiving MIDI

### DISCUSSION THREAD

**Discussion by cbmtrx** [RomanKubiak/ctrlr#352] [Q&A]
Total first timer here so apologies if this is already explained somewhere. I've read through several docs but can't seem to find an answer.

I have a Ctrlr panel (5.3.198) sending MIDI from Logic (Mac Catalina) via an EMU X-MIDI out to a synth, and a MIDI Thru from there to the TX81Z—long story why it's set up this way, but the TX81Z is on channel 2 (the other on ch 1) and MIDI has worked flawlessly for playback/control. Theoretically, sending MIDI from the Ctrlr panel should not be any different...right?

I see MIDI being sent when I adjust values in the panel but there is no response from the TX. I've set all input/output devices to channel 2. Also tried "All" just to get any response.

I saw mentioned somewhere that it's necessary to include a device ID when outputting MIDI otherwise the synth will ignore it (?)... I'm using the pre-built panel so I'm not sure how I would do that.

**Comment by damiensellier** [comment]
Sometimes some synths alter and filter certain types of midi messages on their THRU port.
I would suggest that first, you check if your TX81Z receives and behaves properly from the Ctrlr Pannel if it's just wired straight from X-MIDI OUT to TX81Z  both on channel 1. 
Starting from that, adjust to channel 2 for both the midi card and the TX81Z, and so on, (go through the other synth on CH1, then CH2) until you find the weak link.
Let us know if it's working wired directly from your midi usb out to the TX81Z ;)

**Comment by tomlem** [comment]
A Device ID is _a kind of a MIDI channel_ that is used to send **sysex messages**, not for other MIDI messages such as control change messages, channel voice or channel mode messages.
In this way, sysex messages can be sent to one or more MIDI devices, but not to all of them in the chain. Each device can therefore receive separate commands depending on the set Device ID even if all MIDI devices in the chain are of the same model and same manufacturer. 
This is the reason why this kind of message is called an _exclusive_ message, and this system is called system _exclusive_ or in short sysex. 
Every sysex command must therefore also contain a MIDI channel of the device(s) to be addressed.
So: 
>Check the set Device ID on each of your devices. E.g. -> 'SysEx Device ID=17'

>Check if your device has a setting to turn receiving sysex on or off. Be sure it is set to on. E.g. -> 'Rx SysEx = On'

>Check which Device ID is built into your sysex messages. For this you will need the MIDI implementation chart of your device. E.g. for the Roland U220 the third byte in the sysex message indicates the Device ID. And, after figuring out which byte in your type of device is the ID byte, modify this byte in the sysex command if needed.

(To avoid any confusion: note that the 'SysEx Device ID Channel' is distinguishable from a standard 'Receive Channel' used to send or receive channel voice messages as well from the 'Control Channel' on some devices. E.g. on the Roland U220 the latter specifies the channel on which program changes will be received to select patches and control change data to modify effect parameters.)


If you bought a ready-made Ctrlr panel, your panel manual should mention the ID used. 
A well-built commercial panel should actually allow you to change the Device ID for any sysex command in a user-friendly way.
If your panel was a gift, you should remember that it is not polite to look a gift horse in the mouth; nevertheless, you can try to take a look at the sysex commands used and, after figuring out which byte for your type of synth is the Device ID byte, modify this byte in the sysex command(s) if needed.

It may well be that your Ctrlr panel sends data such as velocity levels, receive channels, key ranges, output assigns and so on mainly using sysex messages.

**Comment by cbmtrx** [comment]
Thanks all. Guess I'll wait to hook it up directly; didn't know if the thru was interfering.

I'm just using the available TX81Z panel from the Ctrlr site.

**Reply by Youngwhisk** [reply]
I am working on some things for my TX81Z as well. You haven't said, but System Exclusive must be turned on under the TX81Z Util Menu. Also you must have the Midi Output device set in Ctrlr to the correct interface you are using. Also the output channel. You can send individual messages from the bottom section of the Console window in the Lua Editor. Those messages must include the channel in the correct format. (1 as the 1st character and the channel# - 1 in hex format for the second character.

An example to set Poly or Mono mode:

F0 43 10 12 3F 00 F7 

All System Exclusives start with F0 and end with F7

The 43 indicates it is a Yamaha product. There is no device number for the TX81Z. It uses channel numbers.

The 3rd set (10) is the channel number. 10 = channel 1.  11 is channel 2 , 1F is channel 16. This must be set correctly or there will be no response from the TX81Z. The channel info in the TX81Z system Exclusive starts at 0 for Channel 1

The 12 and 3F determine the parameter to change.

Changing the 6th set between 00 and 01 will toggle beween Poly and Mono.

As long as the Midi Output Device is set correctly in Crlr and you set the Midi Output channel correctly in this (F0 43 10 12 3F 00 F7)  in the bottom section of the Ctrlr Console window , the TX81Z display should show the changes.

All you need to do is type in the correct code and hit the enter button.

**Reply by cbmtrx** [reply]
Thanks for the tip on enabling sysex on the TX. I wouldn't have known that!

**Comment by cbmtrx** [comment]
OK got it working nicely on the TX81Z. One thing I noticed is a MIDI SYSEX BUFFER FULL message on the TX if I turn a knob too fast. Any way around this?

Super handy tool (considering the unfriendly Yamaha interface).

**Comment by tomlem** [comment]
Of course the easiest solution is turning not that fast.:-)

There are other solutions that are not that easy at all. 

E.g. you can construct a lua routine that causes the value of the button to be sent to the synth only when you tap with the mouse in the value field next to the already set button.

E.g. you can enter the desired value of a slider directly in the slider's numerical field.

@RomanKubiak: it would be great if there was a standard built-in feature that causes the value of the sliders or knobs only to be sent to the synth when the sliding or turning stops.

**Reply by damiensellier** [reply]
Do you mean sending the message only on mouse-up? 
In ctrlr there's Mouse-Down, Mouse-Drag, Mouse-Move and double-click available.

![Ctrlr Mouse event listeners properties](https://user-images.githubusercontent.com/55164385/134392473-ddff464b-68de-4c51-9b50-15e3d34f793c.png)


There's a Mouse-Up event listener in JUCE not sure if it could be implemented to Ctrlr, but that would be really nice to release the message only on release for sure : https://docs.juce.com/master/classMouseListener.html#ac5eb93de1cfd68a35473071df3e2e8cc

I think @godlike-productions and @dnaldoog found a workaround for the same "Dial" feature on his Kurzweil Panel : 
https://github.com/RomanKubiak/ctrlr/discussions/228

We also talked a lot previously about the overflow of messages sent from the panel to the devices, it's here https://github.com/RomanKubiak/ctrlr/discussions/230. A way to work with that also is to use timers and "sample and hold" the value of the knob for a reasonable amount of time for the device to limit the flow. 
If you need an interval of 500ms between messages you have to run a timer to get the value of the knob only every +500ms then send it.

The other way depends on the device, some units send "ready" sysex messages. So you can create a statement if last received message is "ready" then send the message. I used it on my Ensoniq panel and it works very well. by default ReadyState == 0 and when you receive the "ready" message, ReadyState == 1, you send the message based on this condition and reset ReadyState to  == 0 everytime.
I don't know if your TX sends sysex "ready" messages after the previous message has been processed though.

Good luck

**Reply by cbmtrx** [reply]
I'm just using a pre-built panel for the TX (not enough free time to look under the hood yet). tomlem's suggestion is basically what I'm doing—turning the dials slow!

**Comment by tomlem** [comment]
@damiensellier:
_Do you mean sending the message only on mouse-up?_
Yes :-)
